### ISSUE TITLE
[FEATURE] Add iam_role_arn for databricks_group_instance_profile resource

### ISSUE THREAD

**Issue by viktor-legachev-quantori** [databricks/terraform-provider-databricks#1710]
### Use-cases

In noticeable amount of cases instance profile name and associated IAM role have different names. Currently `databricks_group_instance_profile` resource supports only `instance_profile_id` input and based on the profile name it generates associated role with the same name. 

### Attempted Solutions

Set up correct association within Admin UI on Databricks instance profiles page. But it's not acceptable in IaC paradigm.

### Proposal

Add associated IAM role ARN input into the `databricks_group_instance_profile` resource.

**Comment by nfx** [comment]
does https://registry.terraform.io/providers/databricks/databricks/latest/docs/resources/group_role solve your feature request?

**Comment by viktor-legachev-quantori** [comment]
@nfx

> does https://registry.terraform.io/providers/databricks/databricks/latest/docs/resources/group_role solve your feature request?

Not sure, seems that doesn't, I'd like to set up explicit association of Instance profile and IAM role within Databricks admin console: 
![image](https://user-images.githubusercontent.com/101628039/199072836-5d359123-8b9f-42d3-b03a-02ab20fb7d80.png)

**Comment by Jorge-Rodriguez** [comment]
I can confirm that this is valid and interferes with the deployment of instance profiles for use with SQL serverless.

The databricks REST API for instance profiles allows to specify both the instance profile and role ARNs and is properly documented. The `databricks_instance_profile` resource does not conform to the API.

The suggestion by @nfx does not address this issue.

I would suggest rewording this issue's title as:
[BUG] `databricks_instance_profile` does not support the `iam_role_arn` parameter.

For reference, here's the [relevant API documentation](https://docs.databricks.com/dev-tools/api/latest/instance-profiles.html#request-structure)
